packages/server/src/github/api.js

```diff
@@ -0,0 +1,231 @@
+const assert = require('assert');
+const _ = require('lodash');
+const request = require('request');
+const crypto = require('crypto');
+
+const config = require('../config');
+
+class API {
+  /**
+   * @public
+   */
+  constructor() {
+    this.repoOwner = config.GITHUB_REPO_OWNER;
+    this.repoName = config.GITHUB_REPO_NAME;
+    this.token = config.GITHUB_PERSONAL_ACCESS_TOKEN;
+    this.webhookSecret = config.GITHUB_WEBHOOK_SECRET;
+    this.apiURL = `https://api.github.com`;
+    this.serverURL = config.SERVER_URL;
+    this.hookUrl = `${this.serverURL}/hooks`;
+    this.isReady = true;
+
+    if (!this.repoOwner) {
+      console.warn(`GitHub repo owner is not set.`);
+      this.isReady = false;
+    }
+    if (!this.repoName) {
+      console.warn(`GitHub repo name is not set.`);
+      this.isReady = false;
+    }
+    if (!this.token) {
+      console.warn(
+        `GitHub personal access token is not set, ` +
+        `see https://github.com/settings/tokens.`
+      );
+      this.isReady = false;
+    }
+    if (!this.webhookSecret) {
+      console.warn(`GitHub webhook secret is not set`);
+      this.isReady = false;
+    }
+  }
+
+  /**
+   * @protected
+   * @param {string} uri
+   * @param {string} method
+   * @param {null | object} body
+   * @return {Promise<object>}
+   */
+  async _query(uri, method = 'GET', body = null) {
+    assert(_.isString(uri), 'URI should be a string');
+    assert(_.isString(method), 'Method should be a string');
+    assert(_.isObject(body) || _.isNull(body), 'Method should be a string');
+
+    const options = {
+      uri,
+      method,
+      body,
+      headers: this._getHeaders(),
+      json: true,
+      followAllRedirects: true
+    };
+    // console.log('GitHut query options', options);
+    return new Promise((resolve, reject) => {
+      request(options, (error, response, body) => {
+        if (error) return reject(error);
+        resolve(body)
+      });
+    });
+  }
+
+  /**
+   * @protected
+   * @return {{Authorization: string, Accept: string, "User-Agent": string}}
+   */
+  _getHeaders() {
+    assert(_.isString(this.token),
+      'GitHub personal access token should be a string');
+    return {
+      'Authorization': `bearer ${this.token}`,
+      'User-Agent': 'Foretold App',
+      'Accept': 'application/vnd.github.v3+json',
```

Comment:
I'd also want these to be pulled out

Reply:
What exactly here?

Reply:
Sorry; the most important bits are the "Foretold App" and 'application/vnd.github.v3+json' strings

Reply:
Fixed.